src/generalizedlinearmixedmodel.jl

```diff
@@ -529,6 +534,52 @@ ranef(m::GeneralizedLinearMixedModel; uscale::Bool=false) = ranef(m.LMM, uscale=
 
 LinearAlgebra.rank(m::GeneralizedLinearMixedModel) = first(m.LMM.feterms).rank
 
+"""
+    refit!(m::GeneralizedLinearMixedModel[, y::Vector];
+          fast::Bool = (length(m.θ) == length(m.optsum.final)), 
+          nAGQ::Integer = m.optsum.nAGQ))
+
+Refit the model `m` after installing response `y`.
+
+If `y` is omitted the current response vector is used.
+
+If not specified, the `fast` and `nAGQ` options from the previous fit are used.
+
+"""
+function refit!(m::GeneralizedLinearMixedModel{T};
+                fast::Bool = (length(m.θ) == length(m.optsum.final)), 
+                nAGQ::Integer = m.optsum.nAGQ)  where T
+    
+    deviance!(m, 1)
+    reevaluateAend!(m.LMM)
+    
+    reterms = m.LMM.reterms
+    optsum = m.LMM.optsum
+    # we need to reset optsum so that it
+    # plays nice with the modifications fit!() does 
+    optsum.lowerbd = mapfoldl(lowerbd, vcat, reterms)
+    optsum.initial = mapfoldl(getθ, vcat, reterms)
+    optsum.final = copy(optsum.initial)
+    optsum.xtol_abs = fill!(copy(optsum.initial), 1.0e-10)
+    optsum.initial_step = T[]
+    optsum.feval = -1
+
+    fit!(m; fast=fast, nAGQ=nAGQ)
+end
+
+function refit!(m::GeneralizedLinearMixedModel{T}, y; 
+                fast::Bool = (length(m.θ) == length(m.optsum.final)), 
+                nAGQ::Integer = m.optsum.nAGQ) where T
+    resp = m.y
```

Comment:
I think this is a bit confusing to call this `resp` as `m.y` actually evaluates to `m.resp.y`.  It seems that there are too many `resp`s in play here.  Perhaps `mrespy` or `m_resp_y`?

I would also make the rhs more explicit as `m.resp.y`.  I had forgotten about `getproperty` and initially looked for a field named `y` in `GeneralizedLinearMixedModel`.

Reply:
Ah, good points all. I'll modify and update.